src/com/amazon/ion/BufferConfiguration.java

```diff
@@ -133,10 +130,10 @@ public final DataHandler getDataHandler() {
         }
 
         /**
-         * Set the maximum number of bytes between top-level values. This can be used to limit growth of the internal
-         * buffer. For binary Ion, the minimum value is 5 because all valid binary Ion data begins with a 4-byte Ion
-         * version marker and the smallest value is 1 byte. For delimited text Ion, the minimum value is 2 because the
-         * smallest text Ion value is 1 byte and the smallest delimiter is 1 byte. Default: Integer.MAX_VALUE.
+         * Set the maximum size of the buffer. For binary Ion, the minimum value is 5 because all valid binary Ion data
+         * begins with a 4-byte Ion version marker and the smallest value is 1 byte. For text Ion, the minimum value is
+         * 2 because the smallest text Ion value is 1 byte and the smallest delimiter is 1 byte.
+         * Default: Integer.MAX_VALUE.
```

Comment:
Is there any way for a user to specify no limit? Is `Integer.MAX_VALUE` a hard limit?

Reply:
`Integer.MAX_VALUE` is a hard limit because the reader uses a single `byte[]` as its internal buffer, and `byte[]` in Java is indexed with an `int`. When continuable mode is enabled, this limits the size of top-level values to 2GB, which is why we could not make continuable mode default. When continuable mode is disabled, this only limits the size of *scalar values that the user requests to parse* to 2GB. This is a limitation we're more comfortable with.